src/backend/catalog/index.c

```diff
@@ -2457,11 +2457,9 @@ IndexBuildScan(Relation parentRelation,
 	 * concurrent build, or during bootstrap, we take a regular MVCC snapshot
 	 * and index whatever's live according to that.
 	 *
-	 * If the relation is an append-only table, we also use a regular MVCC
-	 * snapshot.
+	 * If the relation is an append-only table, we use SnapshotAny.
 	 */
-	if (IsBootstrapProcessingMode() || indexInfo->ii_Concurrent ||
-			 RelationIsAppendOptimized(parentRelation))
+	if (IsBootstrapProcessingMode() || indexInfo->ii_Concurrent)
```

Comment:
I suspected that we can have `indexInfo->ii_Concurrent == true` under some conditions, but it seems that we never get to this code sections asl GPDB [ignores](https://github.com/arenadata/gpdb/blob/8e2bd7b465aa9dc059ff9956d27bfd3ceca02d11/src/backend/commands/indexcmds.c#L1088) `create index concirrently`.